module/checker.go

```diff
@@ -254,9 +257,21 @@ func (m *Module) CheckEnum(ft FieldType, r *validate.EnumRules) {
 	}
 }
 
-func (m *Module) CheckMessage(ft FieldType, r *validate.MessageRules) {
-	if !r.GetSkip() {
-		m.CheckRules(m.mustFieldType(ft).Embed())
+func (m *Module) CheckMessage(f pgs.Field, rules *validate.FieldRules) {
+	if emb := f.Type().Embed(); emb != nil && emb.IsWellKnown() {
+		if emb.WellKnownType() == pgs.AnyWKT {
+			m.Failf("Any rules should be used for Any fields")
+		}
+		if emb.WellKnownType() == pgs.DurationWKT {
+			m.Failf("Duration rules should be used for Duration fields")
+		}
+		if emb.WellKnownType() == pgs.TimestampWKT {
+			m.Failf("Timestamp rules should be used for Timestamp fields")
+		}
+	}
+
+	if rules.Type != nil && rules.GetMessage().GetSkip() {
```

Comment:
I don't think this will do what you want it to. What does `rules.Type != nil` do?

Reply:
Discussed offline that `rules.Type != nil` checks if any field rules have been set. So this would ensure other rules are not used with `Skip`